_rules/aria-required-owned-element-bc4a75.md

```diff
@@ -0,0 +1,283 @@
+
+## Accessibility Support
+
+This rule relies on assistive technologies to recognize owned elements, as defined by [WAI-ARIA 1.1](https://www.w3.org/TR/wai-aria). This includes when they are nested descendants that are not immediate children. However, some assistive technologies do not recognize owned elements that are not immediate children, unless workarounds are used.
+
+## Background
+
+- [Required Owned Element](https://www.w3.org/TR/wai-aria-1.1/#mustContain)
+- [Owned Element](https://www.w3.org/TR/wai-aria-1.1/#dfn-owned-element)
+- [HTML in ARIA](https://www.w3.org/TR/html-aria/)
+- [Implicit WAI-ARIA Semantics](https://www.w3.org/TR/wai-aria-1.1/#implicit_semantics)
+
+## Test Cases
+
+### Passed
+
+#### Passed example 1
+
+Element `ul` with role `list` has at least one child node `span` with the required owned element `listitem` as an explicit semantic role.
+
+```html
+<ul role='list'>
+	<span role='listitem'>
+	</span>
+	<span>
+	</span>
+</ul>
+```
+
+#### Passed example 2
+
+Element `div` with role `tablist` has child node `span` with the required owned element `tab` as an explicit semantic role.
+
+```html
+<div role='tablist'>
+	<span role='tab'>
+	</span>
+</div>
+```
+
+#### Passed example 3
+
+Multiple roles that each have an instance of their required owned elements present as child nodes.
+
+```html
+<table role='grid'>
+	<tr role='row'>
+		<span role='cell'>
+		</span>
+	</tr>
+</table>
+```
+
+#### Passed example 4
+
+Multiple types of required owned elements are present.
+
+```html
+<ul role='menu'>
+	<li></li> <!-- implicit role -->
+	<li role='menuitem'></li>
+	<li role='menuitemradio'></li>
+</ul>
+```
+
+#### Passed example 5
+
+Required owned element does not have to be an immediate child of the role, but could be any descendant. (See accessibility support notes for details).
+
+```html
+<div role="table">
+	<div>
+		<div>
+			<div role="row">
+				<div role="cell">my cell</div>
+				<div role="cell">and another cell</div>
+			</div>
+			<div role="row">
+				<div role="cell">my cell</div>
+				<div role="cell">and another cell</div>
+			</div>
+		</div>
+	</div>
+</div>
+```
+
+#### Passed example 6
+
+Nested required owned elements.
+
+```html
+<div role="grid">
+	<div role="rowgroup">
+		<div role="row">
+			<div role="gridcell">1</div>
+			<div role="gridcell">Item 1</div>
+			<div role="gridcell">2 Nos.</div>
+		</div>
+	</div>
+	<div role="rowgroup">
+		<div role="row">
+			<div role="gridcell">1</div>
+			<div role="gridcell">Item 1</div>
+			<div role="gridcell">2 Nos.</div>
+		</div>
+	</div>
+</div>
+```
+
+#### Passed example 7
+
+Element with role `list` contains a required owned element which has an implicit role of `listitem`.
+
+```html
+<div role='list'>
+   <li>
+   </li>
+</div>
+```
+
+#### Passed example 8
+
+Element with role `menubar` can have multiple required owned element (`menuitem` `menuitemcheckbox` and `menuitmeradio`) but only one is necessary.
```

Comment:
I think it would be a bit better if the example contained 2 of the three possible elements but not all of them.

The reason is that an incorrect tool that would test for the presence of *all* of them would incorrectly fail on a test case with only some of them and thus we could detect such a problem.
It also makes it a bit easier (I think) for humans to understand "we don’t need all of them" if we show them that it's fine to have only some.